[CIVP-19741] Add _check_column_types method

zacharydfriedlander opened the pull request:
We can use this method to determine when there is a mismatch between column types or counts. Currently it raises a `CivisImportError`, but one we have completed other work on the Civis Platform to expose `loosen_types` in the `imports/files/csv`API we can refactor as we'd like.

Commit:
Add _check_column_types method to determine when there is a mismatch between column types or counts

Commit:
Refactor _process_cleaning_results tests to look for non-columns errors

jkulzick commented on civis/io/_tables.py:
```diff
@@ -1335,3 +1329,58 @@ def _process_cleaning_results(cleaning_futures, client, headers,
+        # share a base type (e.g. INT, VARCHAR, DECIMAl) rather than that
+        # they have the same precision and length
+        # (e.g VARCHAR(42), DECIMAL(8, 10))
+        tcol_base_type = tcol['sql_type'].split('(', 2)[0]
```
nit, but I think `.split('(', 1)` will suffice for both of these. It appears that the number of elements returned is actually `maxsplit + 1`. See [here](https://docs.python.org/3/library/stdtypes.html#str.split).

zacharydfriedlander replied:
You are correct - I was misremembering the parameter's meaning. I thought it was the maximum number of pieces into which to break. Reduced to 1.

jkulzick reviewed (changes requested):
One minor comment

Commit:
Reduce maxsplit to 1

Since maxsplit is the number of breaks (not the number of pieces) we only need 1.

jkulzick reviewed (approved):
LGTM %Travis
